src/main/java/it/gov/pagopa/pu/sil/connector/send_notification/config/SendNotificationApisHolder.java

```diff
@@ -0,0 +1,50 @@
+package it.gov.pagopa.pu.sil.connector.send_notification.config;
+
+import it.gov.pagopa.pu.sendnotification.controller.ApiClient;
+import it.gov.pagopa.pu.sendnotification.controller.BaseApi;
+import it.gov.pagopa.pu.sendnotification.controller.generated.NotificationApi;
+import it.gov.pagopa.pu.sil.config.RestTemplateConfig;
+import jakarta.annotation.PreDestroy;
+import org.springframework.boot.web.client.RestTemplateBuilder;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestTemplate;
+
+@Service
+public class SendNotificationApisHolder {
+
+    private final NotificationApi notificationApi;
+
+    private final ThreadLocal<String> bearerTokenHolder = new ThreadLocal<>();
+
+    public SendNotificationApisHolder(
+        SendNotificationApiClientConfig clientConfig,
+        RestTemplateBuilder restTemplateBuilder
+    ) {
+        RestTemplate restTemplate = restTemplateBuilder.build();
+        ApiClient apiClient = new ApiClient(restTemplate);
+        apiClient.setBasePath(clientConfig.getBaseUrl());
+        apiClient.setBearerToken(bearerTokenHolder::get);
+        apiClient.setMaxAttemptsForRetry(Math.max(1, clientConfig.getMaxAttempts()));
+        apiClient.setWaitTimeMillis(clientConfig.getWaitTimeMillis());
+        if (clientConfig.isPrintBodyWhenError()) {
+          restTemplate.setErrorHandler(RestTemplateConfig.bodyPrinterWhenError("AUTH"));
```

Comment:
This string is a label associated to the particular REST_INVOKE log: it should be the name of the application integrated
```suggestion
          restTemplate.setErrorHandler(RestTemplateConfig.bodyPrinterWhenError("SEND_NOTIFICATION"));
```